src/main.rs

```diff
@@ -48,40 +66,55 @@ fn main() {
         .setting(AppSettings::SubcommandRequired)
         .get_matches();
 
-    println!("Welcome to a round of 'battleship'");
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    ///                                         Start                                           ///
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    println!("{}", Yellow.paint("Welcome to a round of 'battleship'"));
 
     // default --> player vs. player
-    let mut mode = types::Mode::PvP;
+    let mut mode = Mode::PvP;
 
     match battleship.subcommand() {
         ("server", Some(server_args)) => {
             // required arguments
             let port = validate_port(server_args.value_of("port").unwrap());
-            let name = server_args.value_of("name").unwrap();
+            let name = String::from(server_args.value_of("name").unwrap());
 
             // optional arguments
             let mut size = BOARD_SIZE;
             if let Some(val) = server_args.value_of("size") {
+                println!(
+                    "{}",
+                    Red.paint("NOT IMPLEMENTED: Custom board size will be reset to 10x10")
+                );
                 match val.parse::<u8>() {
                     Ok(val) => size = val,
-                    Err(_) => size = helper::read_usize() as u8,
+                    Err(_) => size = util::read_usize() as u8,
                 }
             }
 
+            let mut board = ::model::types::Board::init();
+            if let Some(b) = server_args.value_of("board") {
+                board = util::read_extern_board(b);
+            }
+
             if let Some(_) = server_args.value_of("ships") {
-                println!("NOT IMPLEMENTED: loading custom ship configuration");
+                println!("{}", Red.paint("NOT IMPLEMENTED: Custom ship configuration"));
             }
 
             println!(
-                "create server-player: '{}' -- listening on port: {} -- {}x{} board",
-                name,
+                "create server-player: '{}' -- connecting to port: {} -- {2}x{2} board",
+                &name,
                 port,
                 size,
-                size,
             );
 
-            // TODO: create server instance
+            // create server
+            let wait = thread::spawn(move || server::init(name, size, board));
```

Comment:
I just noticed: I think you don't need a new thread here anyway. When I told you about threads I had something else in mind. Should work fine in the main thread as well.